components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/UserSessionDAO.java

```diff
@@ -29,4 +30,12 @@
 
     UserSession getSession(String sessionId) throws SessionManagementServerException;
 
+    /**
+     * Get federated authentication session mapping info during the federated idp init logout
+     *
+     * @param fedIdpSessionId
+     * @return A map containing federated authentication session details
+     * @throws SessionManagementServerException
+     */
+    FederatedUserSession getFederatedAuthSessionDetails(String fedIdpSessionId) throws SessionManagementServerException;
```

Comment:
have the accessmodifier as default.
`default FederatedUserSession getFederatedAuthSessionDetails(..)` . If we don't have this modifier, if someone extended this interface, they might get issues when adding new public API in an interface

Reply:
It exceeds 120 line width